### PR TITLE
Delete archival config from cluster metadata

### PR DESCRIPTION
**Pull request by yycptt**

### CODE REVIEW

File: common/archiver/archivalMetadata.go [hunk 1]
```diff
@@ -103,3 +156,21 @@ func getDomainArchivalStatus(str string) (shared.ArchivalStatus, error) {
 	}
 	return shared.ArchivalStatusDisabled, fmt.Errorf("invalid archival status of %v for domain, valid status are: {\"\", \"disabled\", \"enabled\"}", str)
 }
+
+// GetTestArchivalMetadata return an archival metadata instance for test use
```
Comment (andrewjdawson2016) [inline common/archiver/archivalMetadata.go]: This should exist in testing code file?

File: common/archiver/archivalMetadata.go [hunk 2]
```diff
@@ -51,13 +63,54 @@ const (
...
+		visibilityConfig: NewArchivalConfig(
+			visibilityStatus,
+			visibilityReadEnabled,
+			domainDefaults.Visibility.Status,
+			domainDefaults.Visibility.URI,
+		),
+	}
+}
+
+func (metadata *archivalMetadata) GetHistoryConfig() *ArchivalConfig {
```
Comment (andrewjdawson2016) [inline common/archiver/archivalMetadata.go]: Hummm.... its not actually true that ArchivalConfig is immutable the way we are returning it and exporting fields. What was the reason you wanted to remove the getters and return a reference to struct directly?

File: common/service/service.go [hunk 3]
```diff
@@ -109,6 +111,8 @@ type (
 		messagingClient        messaging.Client
 		dynamicCollection      *dynamicconfig.Collection
 		dispatcherProvider     client.DispatcherProvider
+		archivalMetadata       archiver.ArchivalMetadata
```
Comment (andrewjdawson2016) [inline common/service/service.go]: This looks nice - good idea.

File: common/archiver/archivalMetadata.go (hunk 1 above)
Comment (yycptt) [reply common/archiver/archivalMetadata.go]: Removed.

File: common/archiver/archivalMetadata.go (hunk 2 above)
Comment (yycptt) [reply common/archiver/archivalMetadata.go]: As discussed, created an interface and provided getter methods.

**Pull request merged**